cwlavro-tools/src/main/java/io/cwl/avro/CWL.java

```diff
@@ -145,42 +146,114 @@ public static Object getStub(final Object type, final String value) {
+     * @param json
+     * @param sequenceSafeGson
+     * @param c
+     * @param objectType
+         * @return Collection of objects of Class c
+         */
+    private static Collection<Object> gsonBuilderHelper(JsonElement json, Gson sequenceSafeGson, Class c, boolean objectType) throws
+            GsonBuildException {
+        Collection<Object> objectCollection = new ArrayList<>();
+        if (json.isJsonArray()) {
+            for (final JsonElement jsonElement : json.getAsJsonArray()) {
+                final Object o = sequenceSafeGson.fromJson(jsonElement, c);
+                objectCollection.add(o);
+            }
+            return objectCollection;
+        } else if (json.isJsonObject()){
+            Map<String, Object> map = sequenceSafeGson.fromJson(json.getAsJsonObject(), new TypeToken<Map<String,Object>>() {}.getType());
+
+            try {
+                Object item;
+                Method setId;
+                Method setSource = null;
+                Method setType = null;
+                setId = c.getMethod("setId", CharSequence.class);
+                if (objectType) {
+                    setType = c.getMethod("setType", Object.class);
+                } else {
+                    setSource = c.getMethod("setSource", Object.class);
+                }
+                for (Map.Entry<String, Object> itemObj : map.entrySet()) {
+                    if (itemObj.getValue() instanceof Map) {
+                        String itemJson = sequenceSafeGson.toJson(itemObj.getValue());
+                        item = sequenceSafeGson.fromJson(itemJson, c);
+                    } else {
+                        item = c.newInstance();
+                        if (objectType) {
+                            setType.invoke(item, itemObj.getValue());
+                        } else {
+                            setSource.invoke(item, itemObj.getValue());
+                        }
+                    }
+                    setId.invoke(item, itemObj.getKey());
+                    objectCollection.add(item);
+                }
+            } catch (InstantiationException ex) {
+                ex.printStackTrace();
+                throw new GsonBuildException("Error casting to class " + c.getName());
+            } catch (IllegalAccessException ex) {
+                ex.printStackTrace();
+                throw new GsonBuildException("Error casting to class " + c.getName());
+            } catch (NoSuchMethodException ex) {
+                ex.printStackTrace();
+                throw new GsonBuildException("No matching methods for class " + c.getName());
+            } catch (InvocationTargetException ex) {
+                ex.printStackTrace();
+                throw new GsonBuildException("Error running method for class " + c.getName());
+            }
+
+            return objectCollection;
+
+        } else{
+            throw new RuntimeException("Unexpected JSON entry");
```

Comment:
This probably should be handled some other way. Perhaps a null return to indicate that the wrong kind of JSON element was passed or a second exception.